### PR TITLE
POAP-152 POAP-151 POAP-150 POAP-149 POAP-148 POAP-147

### PR DESCRIPTION
**Pull request by nahu90**

### CODE REVIEW

File: server/src/routes.ts [hunk 1]
```diff
@@ -904,6 +906,7 @@ export default async function routes(fastify: FastifyInstance) {
   fastify.get(
     '/events',
     {
+      preValidation: [fastify.optionalAuthenticate, ],
```
Comment (ragonzal) [inline server/src/routes.ts]: This endpoint should not include any validation. API is open for the community and other services (such as Etherscan, OpenSea). We should not limit the response here

File: db/schema.sql [hunk 2]
```diff
@@ -97,7 +98,8 @@ CREATE TABLE notifications (
 CREATE TABLE event_host (
     "id" SERIAL PRIMARY KEY,
     "user_id" varchar(256) UNIQUE,
-    "is_active" boolean default true
+    "is_active" boolean default true,
+    "passphrase" varchar(256) UNIQUE
```
Comment (ragonzal) [inline db/schema.sql:102]: Please add the `ALTER TABLE` command in the `db/update-schema.sql`

File: server/src/routes.ts [hunk 3]
```diff
@@ -1881,19 +1914,116 @@ export default async function routes(fastify: FastifyInstance) {
...
+    async (req: any, res) => {
+      const qrCodeHashes:string[] = req.body.qr_list;
+      const numericList:number[] = req.body.numeric_list;
+      let eventId = req.body.event_id || null;
+      let hashesToAdd: any[] = [];
+      let existingClaimedQrs:string[] = [];
+      let existingNumericIds:number[] = [];
+
+      if(numericList && qrCodeHashes.length != numericList.length) {
+        return new createError.BadRequest('qr_code_hashes list lenght is not equal to numeric_list list lenght');
```
Comment (ragonzal) [inline server/src/routes.ts]: Typo in `length`

File: server/src/routes.ts [hunk 4]
```diff
@@ -1881,19 +1914,116 @@ export default async function routes(fastify: FastifyInstance) {
...
+      }
+
+      const checkDuplicatedQrs = qrCodeHashes.filter((item, index) => qrCodeHashes.indexOf(item) != index)
+      const checkDuplicatedNumericIds = numericList ? numericList.filter((item, index) => numericList.indexOf(item) != index) : ""
```
Comment (ragonzal) [inline server/src/routes.ts]: If `numericList` is false, shouldn't the result be an empty array?

File: server/src/routes.ts [hunk 5]
```diff
@@ -1881,19 +1914,116 @@ export default async function routes(fastify: FastifyInstance) {
...
+
+      if(checkDuplicatedQrs.length > 0) {
+        return new createError.BadRequest('qr_list has duplicated hashes, please fix this and send again: [' + checkDuplicatedQrs + ',]');
```
Comment (ragonzal) [inline server/src/routes.ts]: Let's change the message to 
```
QR Hash list include duplicated codes: ' + checkDuplicatedQrs 
```

File: server/src/routes.ts [hunk 6]
```diff
@@ -1881,19 +1914,116 @@ export default async function routes(fastify: FastifyInstance) {
...
+      }
+      
+      if(checkDuplicatedNumericIds.length > 0) {
+        return new createError.BadRequest('numeric_list has duplicated ids, please fix this and send again: [' + checkDuplicatedNumericIds + ',]');
```
Comment (ragonzal) [inline server/src/routes.ts]: Let's change the message to 
```
Numeric list include duplicated numbers: ' + checkDuplicatedNumericIds
```

**Review by ragonzal — CHANGES_REQUESTED** [review]

File: server/src/routes.ts (hunk 1 above)
Comment (nahu90) [reply server/src/routes.ts]: One of the requirements was that comunity only can see comunity created events and admin is the only one that can see all (comunity + admin created). this is no longer necesary?

Comment (ragonzal) [reply server/src/routes.ts]: Filter will always be made on the frontend app.
We should check on QR assignment (list or range) that the event assigned was created by the community (if the user is not admin)

**Review by ragonzal — APPROVED** [review]

**Pull request merged**